### PR TITLE
Add UI docs

### PR DESCRIPTION
**Pull request by aljesusg**
WIP Add dosc for the UI 

- Add embedded components

Fix issue #183
Require: https://github.com/jaegertracing/jaeger-ui/pull/286

I do not know exactly what kind of documentation would be the most appropriate for this

cc @tiffon 

Demo: https://deploy-preview-185--jaegertracing.netlify.com/docs/next-release/frontend-ui/

### CODE REVIEW

**Comment by aljesusg** [comment]
I called interface instead of UI. WDYT @tiffon ?

Demo: https://deploy-preview-185--jaegertracing.netlify.com/docs/next-release/interface/

**Comment by yurishkuro** [comment]
"interface" is confusing, I would rather call it "Frontend/UI".

File: content/docs/next-release/frontend-ui.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+---
+title: Frontend/UI
+weight: 10
+---
+
+Jaeger Web UI is implemented in Javascript using popular open source frameworks like React. Several performance improvements have been released in v1.0 to allow the UI to efficiently deal with large volumes of data, and to display traces with tens of thousands of spans (e.g. we tried a trace with 80,000 spans).
+
+## Embed Mode
+
+Enhancement to embed mode in Jaeger UI for some components allowing us to integrate it into other applications.
```
Comment (tiffon) [inline content/docs/next-release/frontend-ui.md]: I suggest rephrasing this paragraph to:

-----

>Starting with version **[TODO]**, Jaeger UI provides an "embedded" layout mode which is intended to support integrating Jaeger UI into other applications. Currently (as of `v0`), the approach taken is to remove various UI elements from the page to make the UI better suited for space-constrained layouts.
>
>The embedded mode is induced and configured via URL query parameters.
>
>To enter embedded mode, the `uiEmbed=v0` query parameter and value must be added to the URL. For example, the following URL will show the trace with ID `abc123` in embedded mode:
>
>```
>https://example.com/trace/abc123?uiEmbed=v0
>```
>
>`uiEmbed=v0` is required.
>
>Further, each page supported has an Embed open window button added that will open the non-embedded page in a new tab.

-----
What do you think? 

Here is the markdown:

    Starting with version **[TODO]**, Jaeger UI provides an "embedded" layout mode which is intended to support integrating Jaeger UI into other applications. Currently (as of `v0`), the approach taken is to remove various UI elements from the page to make the UI better suited for space-constrained layouts.

    The embedded mode is induced and configured via URL query parameters.

    To enter embedded mode, the `uiEmbed=v0` query parameter and value must be added to the URL. For example, the following URL will show the trace with ID `abc123` in embedded mode:

    ```
    https://example.com/trace/abc123?uiEmbed=v0
    ```

    `uiEmbed=v0` is required.
    
    Further, each page supported has an Embed open window button added that will open the non-embedded page in a new tab.

File: content/docs/next-release/frontend-ui.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    end=1543921359557000&
+    limit=20&lookback=1h&
+    maxDuration&
+    minDuration&
+    service=jaeger-query&
+    start=1543917759557000&
+    uiEmbed=v0
+```
+
+![Embed Search Traces](/img/frontend-ui/embed-search-traces.png)
```
Comment (tiffon) [inline content/docs/next-release/frontend-ui.md]: The image URLs need `/static/` prepended.

The images will also need to be updated whenever we land on a final layout and design.

File: content/docs/next-release/frontend-ui.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+We have a button in this kind of view that redirect us to the search traces page in the Jaeger Site <img src="/img/frontend-ui/embed-open-icon.png" style="width: 20px; height:20px; display:inline;" alt="Embed open window">.  
+
+##### Optionally
+
+We can use an extra options in the embed mode :
```
Comment (tiffon) [inline content/docs/next-release/frontend-ui.md]: Suggest rephrasing as:

>The following query parameter can be used to configure the layout of the search page:
>
>- `uiSearchHideGraph=1`
>    - Do not display the scatter plot above the search results

File: content/docs/next-release/frontend-ui.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    maxDuration&
+    minDuration&
+    service=jaeger-query&
+    start=1543917759557000&
+    uiEmbed=v0
+```
+
+![Embed Search Traces](/img/frontend-ui/embed-search-traces.png)
+
+We have a button in this kind of view that redirect us to the search traces page in the Jaeger Site <img src="/img/frontend-ui/embed-open-icon.png" style="width: 20px; height:20px; display:inline;" alt="Embed open window">.
```
Comment (tiffon) [inline content/docs/next-release/frontend-ui.md]: This ^ can be removed if it is added to the section above "EmbedMode" intro section.

File: content/docs/next-release/frontend-ui.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+```sh
+<JAEGER UI URL>/search?
+    end=1543921359557000&
+    limit=20&lookback=1h&
+    maxDuration&
+    minDuration&
+    service=jaeger-query&
+    start=1543917759557000&
+    uiEmbed=v0&
+    uiSearchHideGraph=1s
```
Comment (tiffon) [inline content/docs/next-release/frontend-ui.md]: Looks like a typo, `=1s`

File: content/docs/next-release/frontend-ui.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    service=jaeger-query&
+    start=1543917759557000&
+    uiEmbed=v0
+```
+
+![Embed Search Traces](/img/frontend-ui/embed-search-traces.png)
+
+We have a button in this kind of view that redirect us to the search traces page in the Jaeger Site <img src="/img/frontend-ui/embed-open-icon.png" style="width: 20px; height:20px; display:inline;" alt="Embed open window">.  
+
+##### Optionally
```
Comment (tiffon) [inline content/docs/next-release/frontend-ui.md]: Suggest renaming this title to "Configuration options"

File: content/docs/next-release/frontend-ui.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+<JAEGER UI URL>/trace/<TRACE ID>?
+    uiEmbed=v0
+```
+![Embed Trace view](/img/frontend-ui/embed-trace-view.png)
+
+If we have navigated to this view from the search traces page we'll have a button to go back to the results page.
+
+![Embed Trace view](/img/frontend-ui/embed-trace-view-with-back-button.png)
+
+We have a new button in this kind of view that redirect us to the trace page in the Jaeger Site <img src="/img/interface/embed-open-icon.png" style="width: 20px; height:20px; display:inline;" alt="Embed open window">.
```
Comment (tiffon) [inline content/docs/next-release/frontend-ui.md]: This ^ can be removed if it is added to the section above "EmbedMode" intro section.

File: content/docs/next-release/frontend-ui.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+
+##### Optionally
```
Comment (tiffon) [inline content/docs/next-release/frontend-ui.md]: Suggest renaming this title to "Configuration options"

File: content/docs/next-release/frontend-ui.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+
+We can use an extra options in the embed mode :
```
Comment (tiffon) [inline content/docs/next-release/frontend-ui.md]: I suggest rephrasing to the following and adding images after summarizing the query parameters.

>The following query parameters can be used to configure the layout of the trace page:
>
>- `uiTimelineCollapseTitle=1`
>    - The trace header starts out collapsed, which hides the summary and minimap
>- `uiTimelineHideMinimap=1`
>    - Removes the minimap, regardless of whether the trace header is expanded or not
>- `uiTimelineHideSummary=1`
>    - Removes the trace summary information (number of services, etc.), regardless of whether the trace header is expanded or not

File: content/docs/next-release/frontend-ui.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
+---
+title: Frontend/UI
+weight: 10
+---
+
+Jaeger Web UI is implemented in Javascript using popular open source frameworks like React. Several performance improvements have been released in v1.0 to allow the UI to efficiently deal with large volumes of data, and to display traces with tens of thousands of spans (e.g. we tried a trace with 80,000 spans).
```
[the remark this answers is not in the document]
Comment (aljesusg) [reply content/docs/next-release/frontend-ui.md]: Ok, I just move the content in docs/1.8/features/#modern-web-ui but I agree with you about remove the second sentence

**Comment by aljesusg** [comment]
@tiffon I did the changes. I added the images about trace embed that you uploaded in your https://github.com/jaegertracing/jaeger-ui/pull/286 and I added the images to the public/ too. Anyway We should wait until the merge of https://github.com/jaegertracing/jaeger-ui/pull/286

**Comment by tiffon** [comment]
Thanks, @aljesusg, the changes look great! jaegertracing/jaeger-ui#286 is merged. :+1:

**Comment by aljesusg** [comment]
Great !!!

**Comment by yurishkuro** [comment]
Wouldn't it better to consolidate the configuration options from https://www.jaegertracing.io/docs/1.8/deployment/ into this new page as well, and link to it from Deployment?

**Comment by aljesusg** [comment]
As you want, if you think it's better to make the change I can do it but ...I think that it could be confuse for the user have the deployment manual (components) with the documentation about the UI and how to use/work it. What do you think ? @tiffon @yurishkuro

**Comment by tiffon** [comment]
I was thinking these docs would be for users. So, I agree that it could be
confusing to move the UI config docs to this section.

But, linking the other way seems like it would work. If we add a section to
the UI docs called "Administration" (or something), we can link to the UI
config docs in Deployment.

**Comment by yurishkuro** [comment]
> I was thinking these docs would be for users.

@tiffon someone deploying jaeger is also a "user". If you are talking about real end users who actually use the UI to investigate traces, then this document is clearly not for them - why would they care about embedding and the internal API to control the embedding?  In fact, I almost want to argue that having documentation for end users is an anti-pattern - it would mean the UI is poorly design and non-intuitive. End users may need a tutorial mode in the UI that walks them through the all elements.

> Or What was your idea?

@aljesusg my proposal was to move UI section from Deployment to this doc, and only leave a link in the Deployment. I think both configuration and embedding are part of the UI deployment story, and this doc should be a sub-document of the Deployment (similar to how Client Features doc is sub-doc of Client Libraries).

**Comment by tiffon** [comment]
@yurishkuro Yes, I was referring to folks using the UI. You're right, people deploying Jaeger are also users. I would tend to think of them as administrators, but they're definitely users, either way.

>real end users who actually use the UI to investigate traces, then this document is clearly not for them

Great point. I do think the "Jaeger UI" section is the right place for the docs on embedding, though.

>In fact, I almost want to argue that having documentation for end users is an anti-pattern - it would mean the UI is poorly designed and non-intuitive.

I don't see how having documentation means the UI is poorly designed and non-intuitive. For instance, [Gmail has docs](https://support.google.com/mail#topic=7065107) and I'd say it's very easy to use. Same with [GitHub](https://help.github.com/articles/about-stars/).

As an aside: I think there is very little likelihood that documentation would make up for a confusing UI, so I don't see docs as running the risk of enabling a bad UI.

>End users may need a tutorial mode in the UI that walks them through the UI elements.

I'd say step-by-step guides in the UI would be great for introducing features but unmanageable for a full tutorial or introduction to the UI. For that, I think taking inspiration from something like this [Chrome overview](https://www.google.com/chrome/) would be great.

>I think both configuration and embedding are part of the UI deployment story

Seems like integrating the UI into another product is not a sub-topic of deploying Jaeger. For instance, the Studio team (a reference internal to Uber) doesn't need to know anything about deploying Jaeger if they only plan to embed it.

@aljesusg, you're consuming the embed functionality, are you also involved in deploying Jaeger?

**Comment by aljesusg** [comment]
no @tiffon, only consuming from Kiali project

**Comment by aljesusg** [comment]
I moved the docs to deployment @yurishkuro https://deploy-preview-185--jaegertracing.netlify.com/docs/next-release/deployment/#frontend-ui but I think that we should add the index to the frontend docs like the deployment, I saw that in client features is disable but I think that we have so information in the UI docs. What do you think?

File: content/docs/next-release/deployment.md [hunk 11]
```diff
@@ -298,98 +301,8 @@ docker run \
...
-**jaeger-query** serves the API endpoints and a React/Javascript UI.
-The service is stateless and is typically run behind a load balancer, e.g. nginx.
-
-At default settings the query service exposes the following port(s):
-
-Port  | Protocol | Function
------ | -------  | ---
-16686 | HTTP     | **/api/*** endpoints and Jaeger UI at **/**
-16687 | HTTP     | Health check at **/**
-
```
[the remark this answers is not in the document]
Comment (aljesusg) [reply content/docs/next-release/deployment.md]: renamed

File: content/docs/next-release/frontend-ui.md [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+
+**jaeger-query** serves the API endpoints and a React/Javascript UI.
+The service is stateless and is typically run behind a load balancer, e.g. nginx.
+
+At default settings the query service exposes the following port(s):
+
+Port  | Protocol | Function
+----- | -------  | ---
+16686 | HTTP     | **/api/*** endpoints and Jaeger UI at **/**
+16687 | HTTP     | Health check at **/**
```
Comment (yurishkuro) [inline content/docs/next-release/frontend-ui.md]: please don't remove this section above from the deployment doc. It refers to query-service, a microservice in Jaeger, and logically belongs to deployment. The rest of this doc refers to customization / configuration of the UI.

Comment (yurishkuro) [reply content/docs/next-release/frontend-ui.md]: I added a commit with the change that I wanted.

**Comment by aljesusg** [comment]
Yes, the images are final from @tiffon PR with the last changes

**Comment by yurishkuro** [comment]
thanks!

**Comment by aljesusg** [comment]
Thanks @yurishkuro  & @tiffon

**Pull request merged**